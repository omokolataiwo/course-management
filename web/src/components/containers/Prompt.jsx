import React, { Component } from 'react';
import { Prompt } from 'react-router-dom';
import Modal from './Modal';

class PromptDialog extends Component {
  callback = null;

  state = {
    open: false
  };

  componentDidMount() {
    window[Symbol.for('MODAL_UNSAVED_CHANGES')] = this.showModal;
  }

  componentWillUnmount() {
    delete window[Symbol.for('MODAL_UNSAVED_CHANGES')];
  }

  showModal = (callback) => {
    this.callback = callback;
    this.setState({ open: true });
  };

  dialogAction = (proceed) => {
    if (proceed) {
      return this.callback(true);
    }

    return this.setState({ open: false });
  };

  render() {
    const { when } = this.props;
    return (
      <div>
        <Prompt when={when} message={() => 'MODAL_UNSAVED_CHANGES'} />
        {this.state.open && <Modal action={this.dialogAction} />}
      </div>
    );
  }
}

export default PromptDialog;
