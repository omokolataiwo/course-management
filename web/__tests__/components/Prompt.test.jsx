import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Prompt from '../../src/components/containers/Prompt';

const wrapper = shallow(<BrowserRouter><Prompt /></BrowserRouter>).dive();

describe('Prompt Component', () => {
  it('should pop up modal', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
