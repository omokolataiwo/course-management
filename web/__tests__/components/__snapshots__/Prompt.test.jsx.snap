// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt Component should pop up modal 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PromptDialog />
  </Router>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
